cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/TopicConfigurationParameters.java

```diff
@@ -48,12 +58,18 @@ public TopicConfigurationParameters() {
   @Override
   protected void initParameters() throws UnsupportedEncodingException {
     super.initParameters();
-    _topic = ParameterUtils.topic(_request);
-    if (_topic == null) {
-      throw new IllegalArgumentException("Topic to update configuration is not specified.");
+    Pattern topic = ParameterUtils.topic(_request);
+    Short replicationFactor = ParameterUtils.replicationFactor(_request);
+    if (topic != null && replicationFactor != null) {
+      _topicPatternsByReplicationFactor = Collections.singletonMap(replicationFactor, Collections.singleton(topic));
+    } else if (topic == null && replicationFactor == null) {
+      _topicPatternsByReplicationFactor = ParameterUtils.topicPatternsByReplicationFactor(_request);
```

Comment:
`topicPatternsByReplicationFactor` -> `proposedConfigurations` / `proposedConfigs` (i.e. something that applies to all topic configurations)

Reply:
As we address issue #886 , proposed topic configurations will not be mixed in a signal map, we can address this generalization in that patch.